config/config.go

```diff
@@ -454,6 +454,45 @@ func (c *AlertingConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
+// AlertmanagerAPIVersion represents a version of the
+// github.com/prometheus/alertmanager/api, e.g. 'v1' or 'v2'.
+type AlertmanagerAPIVersion string
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (v *AlertmanagerAPIVersion) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*v = AlertmanagerAPIVersion("")
+	type plain AlertmanagerAPIVersion
+	if err := unmarshal((*plain)(v)); err != nil {
+		return err
+	}
+
+	// Default to Alertmanager API v1.
```

Comment:
This isn't how we generally implement defaults.

Reply:
Changed to the default way of doing defaults. Thanks.